### PR TITLE
Removed external references in requests tests.

### PR DESCRIPTION
**Pull request by rcbjBlueMars**
## Description
Remove calls to external endpoints from test suite.

### Testing
* Test suite runs successfully.
* 
### Documentation
* These changes do not modify the documentation.

### CODE REVIEW

File: tests/aiohttp/flask_app/app1.py [hunk 1]
```diff
@@ -50,8 +50,9 @@ async def fetch(url, body, headers):
     async with session.post(url, data=body, headers=headers) as response:
       return await response.text()
 
+
 @server.route("/route2", methods=['POST'])
-def testAPI1():
+def testAPI2():
```
Comment (jcchavezs) [inline tests/aiohttp/flask_app/app1.py:55]: Please use descriptive names for tests e.g. what this test is trying to assert or test. A good example can be found here:

https://github.com/hypertrace/goagent/blob/main/instrumentation/opentelemetry/net/hyperhttp/handler_test.go#L22

File: tests/aiohttp/flask_app/app1.py [hunk 2]
```diff
@@ -68,12 +69,31 @@ def testAPI1():
       response.data = str('{}')
       return response
 
+@server.route("/route1", methods=['POST'])
+def testAPI1():
```
Comment (jcchavezs) [inline tests/aiohttp/flask_app/app1.py:73]: What is this exactly doing?

File: tests/aiohttp/flask_app/app1.py [hunk 3]
```diff
@@ -68,12 +69,31 @@ def testAPI1():
...
+    try:
+      logger.info('Serving request for /route1')
+      loop = asyncio.get_event_loop()
+      response1 = flask.Response(mimetype='application/json')
+      response1.data = str('{ "a": "a", "xyz": "xyz" }')
+      return response1
+    except:
+      logger.error('Failed to initialize postgresql instrumentation wrapper: exception=%s, stacktrace=%s',
```
Comment (jcchavezs) [inline tests/aiohttp/flask_app/app1.py:81]: Is this log right?

File: tests/aiohttp/flask_app/app1.py [hunk 4]
```diff
@@ -68,12 +69,31 @@ def testAPI1():
...
+      response.data = str('{}')
+      return response
+
 @server.teardown_request
 def after_request(response):
   try:
     logger.debug("after_request() called")
     if not ENABLE_INSTRUMENTATION:
       return response
+    if request.base_url != '/route1':
```
Comment (jcchavezs) [inline tests/aiohttp/flask_app/app1.py:94]: Is this right? I am not exactly sure what these tests for but if we are receiving a request to a non expected endpoint we should fail I think?

File: tests/aiohttp/test_aiohttp_1.py [hunk 5]
```diff
@@ -55,11 +55,11 @@ async def test_run():
 
   #Make test call
   async with aiohttp.ClientSession() as session:
-    async with session.post('https://petstore.swagger.io/v2/pet', data='{"id":0,"category":{"id":0,"name":"doggie"},"name":"doggie","photoUrls":["http://example.co"],"tags":[{"id":0,"name":"doggie"}],"status":"available"}', headers={ "Accept":"application/json", "Content-Type": "application/json", 'tester1': 'tester1', 'tester2':'tester2' }) as response:
+    async with session.post('http://localhost:8000/route1', data='{ "a":"b", "c": "d" }', headers={ 'tester1': 'tester1', 'tester2':'tester2' }) as response:
```
Comment (jcchavezs) [inline tests/aiohttp/test_aiohttp_1.py:58]: I'd really prefer to use meaningful values for the tests, this way it is easier to catch dummy issues.

File: tests/aiohttp/tox.ini [hunk 6]
```diff
@@ -28,12 +28,13 @@ commands =
     docker-compose build
     docker-compose rm -v -f
     docker-compose up -d flask_app_b
-    pytest -rPx test_aiohttp_1.py
+    sleep 10
```
Comment (jcchavezs) [inline tests/aiohttp/tox.ini:31]: I'd really avoid this non deterministic sleeps. Can't we use health endpoints?

File: tests/aiohttp/tox.ini [hunk 7]
```diff
@@ -28,12 +28,13 @@ commands =
...
+#    pytest -rPx test_aiohttp_1.py
     pytest -rPx test_aiohttp_2.py
-    pytest -rPx test_aiohttp_3.py
-    docker-compose stop flask_app_b
-    docker-compose up -d flask_app_c
-    pytest -rPx test_aiohttp_4.py
-    docker-compose stop
+#    pytest -rPx test_aiohttp_3.py
```
Comment (jcchavezs) [inline tests/aiohttp/tox.ini:34]: If not use, shall we remove it?

File: tests/requests/flask_app/app2.py [hunk 8]
```diff
@@ -0,0 +1,145 @@
...
+    span_list = memoryExporter.get_finished_spans()
+    # Confirm something was returned.
+    assert span_list
+    # Confirm there are three spans
+    logger.debug('len(span_list): ' + str(len(span_list)))
+    assert len(span_list) == 1
+    logger.debug('span_list: ' + str(span_list[0].attributes))
+    # Convert each span to a JSON object
+    flaskSpanAsObject = json.loads(span_list[0].to_json())
+    # Flask extended span object attributes should look like:
```
Comment (jcchavezs) [inline tests/requests/flask_app/app2.py:76]: This kind of documentation is somehow meaningless in tests. If this is important we should have assertions instead.

File: tests/requests/flask_app/app2.py [hunk 9]
```diff
@@ -0,0 +1,145 @@
...
+  logger.info('Agent initialized.')
+  logger.info('Adding in-memory span exporter.')
+  memoryExporter = InMemorySpanExporter()
+  simpleExportSpanProcessor = SimpleSpanProcessor(memoryExporter)
+  agent.register_processor(simpleExportSpanProcessor)
+  logger.info('Added in-memoy span exporter') 
+
+  # Setup Jaeger Exporter
+  logger.info('Adding jaeger span exporter.')
+  jaegerExporter = JaegerExporter(
```
Comment (jcchavezs) [inline tests/requests/flask_app/app2.py:138]: I am curious about this. Why exactly do we export on jaeger which uses thrift I think? Can't we export over JSON using e.g. zipkin or OTLP?

**Review by jcchavezs — CHANGES_REQUESTED** [review]
Thanks @rcbjBlueMars I added a bunch of comments.

**Comment by rcbjBlueMars** [comment]
The aiohttp tests no longer make exteral calls.

The original intent of this PR has been addressed. I will now address the feedback @jcchavezs provided.

**Comment by jcchavezs** [comment]
Closed as agreed with @rcbjBlueMars

**Pull request closed without merging**